sechub-webui/src/test/java/com/mercedesbenz/sechub/webui/YamlPropertyLoaderFactory.java

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.webui;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.springframework.boot.env.YamlPropertySourceLoader;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.DefaultPropertySourceFactory;
+import org.springframework.core.io.support.EncodedResource;
+import org.springframework.lang.Nullable;
+
+/**
+ * Spring Boot's default property source factory does not support YAML files
+ * directly in `@TestPropertySource`. By using this custom factory, we can load
+ * properties from YAML files, which is often more convenient and readable for
+ * complex configurations.
+ *
+ * @author hamidonos
+ */
+public class YamlPropertyLoaderFactory extends DefaultPropertySourceFactory {
+
+    @Override
+    public PropertySource<?> createPropertySource(@Nullable String name, EncodedResource resource) throws IOException {
+        Resource actualResource = resource.getResource();
```

Comment:
Hm... if the actual resource exists... shouldn't we check if the resource name ends with ".yml" or ".yaml" as well - otherwise it could happen IMHO that this loader factory would try to load also other kind of resources - and fail (but maybe I am wrong).

Ah, have seen it now in usage: its only for tests and defined directly inside tests... so normally not problematic.